include/crpropa/module/PhotoPionProduction.h

```diff
@@ -12,13 +12,20 @@ namespace crpropa {
  * @{
  */
 
+struct SophiaEventOutput {
+	int nParticles;
+	std::vector<double> energy;
+	std::vector<int> id;
```

Comment:
Is the id variable SOPHIA-specific? If so, I suggest to go with a dedicated enum class in `sophia.h` where all particle ids are defined (like I wrote [here](https://github.com/CRPropa/CRPropa3/pull/260#discussion_r363662596)) and then this should be `std:vector<SophiaParticle>`.

```c++
enum class SophiaParticle : int {
   proton = 13,
   antiproton = -13,
   neutron = -14,
   ...
};
```
Instead of `int`s later, you use `SophiaParticle` instead, and if-comparisons are directly readable (no additional comments in the code are needed). For example, `if pType == SophiaParticle::proton`.

I don't intend to make this suggestion as a PR blocker - we can implement it later.

Reply:
It would be clearer to change the ID values in sophia to the HEPID standard as in CRPropa - but this is an issue for another day.